zebra-network/src/meta_addr.rs

```diff
@@ -117,20 +154,27 @@ pub struct MetaAddr {
     ///
     /// `last_seen` times from `NeverAttempted` peers may be invalid due to
     /// clock skew, or buggy or malicious peers.
-    pub last_seen: DateTime<Utc>,
+    pub fn get_last_seen(&self) -> DateTime<Utc> {
+        self.last_seen
+    }
 
-    /// The outcome of our most recent communication attempt with this peer.
-    pub last_connection_state: PeerAddrState,
-}
+    /// Update the last time we interacted with this peer to the current time.
+    pub fn update_last_seen(&mut self) {
+        self.last_seen = Utc::now();
+    }
 
-impl MetaAddr {
-    /// Sanitize this `MetaAddr` before sending it to a remote peer.
-    pub fn sanitize(mut self) -> MetaAddr {
+    /// Return a sanitized version of this `MetaAddr`, for sending to a remote peer.
+    pub fn sanitize(&self) -> MetaAddr {
         let interval = crate::constants::TIMESTAMP_TRUNCATION_SECONDS;
-        let ts = self.last_seen.timestamp();
-        self.last_seen = Utc.timestamp(ts - ts.rem_euclid(interval), 0);
-        self.last_connection_state = Default::default();
-        self
+        let ts = self.get_last_seen().timestamp();
+        let last_seen = Utc.timestamp(ts - ts.rem_euclid(interval), 0);
+        MetaAddr {
+            addr: self.addr,
+            services: self.services,
+            last_seen,
+            // the state isn't sent to the remote peer, but sanitize it anyway
+            last_connection_state: Default::default(),
+        }
```

Comment:
Here is the security risk fix:
- peers in the address book can't be modified because `sanitize` takes `&self`
- new fields must be sanitized because `sanitize` constructs a new `MetaAddr`, listing every field